### DISCUSSION TITLE
Trigger a callback function after successful server action form submission

### DISCUSSION THREAD

**Discussion by ru4ert** [vercel/next.js#61325] [Help]
### Summary

Imagine you have a [Forms with Next.js and Server Actions](https://github.com/vercel/next.js/tree/canary/examples/next-forms):
``` tsx
export const CreateUser = (props: { onSuccess: () => void; }) => {
  const [state, formAction] = useFormState(createNewUser, { errors: {} })
  return <form
    action={formAction}
    className="grid grid-cols-2 gap-4">
   ....
  </form>
}
```

Is there a way to get some sort of "success" callback?

In the returned object i can place a `success: true` property. A `useEffect` can listen on it and can trigger a `onSuccess` fn.

Are there any better ways to trigger smth after successful submission?

**Comment by charlie-ttt** [comment]
Did you manage to find a way? @ru4ert 
Facing the same problem currently
